### DISCUSSION TITLE
Is it possible to create a tab-like experience with Tauri JS by attaching child windows to parent windows?

### DISCUSSION THREAD

**Discussion by sanjaiyan-dev** [tauri-apps/tauri#7333] [Q&A] [answered]
Is it possible to attach the child windows created in Tauri JS (https://tauri.app/v1/guides/features/multiwindow/) to parent windows to give a tab-like feel ?

**Comment by FabianLars** [comment, marked as answer]
not natively, you _can_ simulate it by syncing it on resize and move events but of course that will be horrible UX due to a slight delay.
Better to have that tabbing solely in html/js until tauri has proper support for this imo (no ETA at all)

**Reply by sanjaiyan-dev** [reply]
> not natively, you _can_ simulate it by syncing it on resize and move events but of course that will be horrible UX due to a slight delay.
> Better to have that tabbing solely in html/js until tauri has proper support for this imo (no ETA at all)

I am deeply sorry to bother you, and I apologize for my request. If it's not too much trouble, could you kindly provide an extremely simple code solution for this? I understand if you're busy, and I truly appreciate any help you can offer. (@FabianLars)

**Reply by FabianLars** [reply]
That totally depends on your frontend stack. virtually any frontend framework has tabbing libraries (often part of general component libraries).
If all the tab content is local to your app then it's pretty easy, just follow their docs.
If you want to display remote content not owned by you then it can get tricky. If these sites "allow" you to display them in iframes, you can use said iframe inside the tab items, and if not you're probably out of luck in which case it sounds like you're trying to build a browser or browser-like app which is currently explicitly not support by Tauri.

**Reply by sanjaiyan-dev** [reply]
> That totally depends on your frontend stack. virtually any frontend framework has tabbing libraries (often part of general component libraries).
> If all the tab content is local to your app then it's pretty easy, just follow their docs.
> If you want to display remote content not owned by you then it can get tricky. If these sites "allow" you to display them in iframes, you can use said iframe inside the tab items, and if not you're probably out of luck in which case it sounds like you're trying to build a browser or browser-like app which is currently explicitly not support by Tauri.

Thank you for your reply. I apologize for any confusion. Regarding my plan to create a small browser, I would like to know if it's possible to open the developer tools for each tab using this method: [link to the documentation](https://docs.rs/tauri/1.4.1/tauri/window/struct.Window.html#method.open_devtools).

I greatly appreciate your expertise and guidance in this matter. Your assistance is invaluable to me, and I am sincerely grateful for your time and support. (@FabianLars)

**Reply by FabianLars** [reply]
thanks for the kind words :)

> I apologize for any confusion. Regarding my plan to create a small browser, I would like to know if it's possible to open the developer tools for each tab using this method

No, the devtools are always bound to the webview itself, which means you will only ever have one devtools per window no matter how many tabs are open.

To be completely honest, i think Tauri _in its current state_ is not the best tool for browser-like use-cases and i'd suggest looking for other tools if the content you're hosting in these tabs are websites not under your full control.
But again, we explicitly don't support it so it makes sense it's not easy to do.

**Reply by sanjaiyan-dev** [reply]
> thanks for the kind words :)
> 
> > I apologize for any confusion. Regarding my plan to create a small browser, I would like to know if it's possible to open the developer tools for each tab using this method
> 
> No, the devtools are always bound to the webview itself, which means you will only ever have one devtools per window no matter how many tabs are open.
> 
> To be completely honest, i think Tauri _in its current state_ is not the best tool for browser-like use-cases and i'd suggest looking for other tools if the content you're hosting in these tabs are websites not under your full control.
> But again, we explicitly don't support it so it makes sense it's not easy to do.

🌟 Thank you ever so much for your enlightening explanation! 🌟
